src/ui/components/AppWrapper/AppWrapper.tsx

```diff
@@ -196,11 +196,14 @@ const AppWrapper = (props: { children: ReactNode }) => {
       await AriesAgent.agent.start();
       setIsOffline(false);
     } catch (e) {
-      // @TODO - foconnor: Should specifically catch the error instead of all, but OK for now.
-      // eslint-disable-next-line no-console
-      AriesAgent.agent.bootAndConnect().then(() => {
-        setIsOffline(!AriesAgent.agent.isAgentReady());
-      });
+      // If the error is failed to fetch with signify, we retry until the connection is secured
+      if (/failed to fetch/gi.test((e as Error).message)) {
```

Comment:
I wonder how reliable this is, in case there's another issue with KERIA (e.g. port is open but responding badly).

Maybe it's more reliable if we create a custom error class, and then if there's an error calling `signify.boot` or `signify.connect` we can wrap the error in this custom error and then try to catch that?

Just want to make sure that it's not something else (like basicStorage starting up) that's failing

Reply:
I tried to simulate the downtime of the server by running the local Keria and then turned it off. It only throws me an error message "failed to fetched" and below is the line where the error got thrown like "at signifyClient.boot...". I can catch that also if you want.

Reply:
Yeah in this case the port isn't available but that might not always be the case (e.g. something else running on the port, or more likely: KERIA is up but not working properly, or is just starting up and not quite ready)

But now I'm wondering: in this PR - we call boot/connect and use a try catch to then call your new function `bootAndConnect`. Can we just directly call `bootAndConnect` and remove the first call and this error check?

Reply:
We can. But I will freeze our `init` method here because we are `await` for the `start` to finish before users can see their identifiers. With the current code, if the `start` throws an error, we will retry silently instead of waiting for it.

Reply:
```
that might not always be the case (e.g. something else running on the port, or more likely: KERIA is up but not working properly, or is just starting up and not quite ready)
```
I guess these cases will also throw the "failed to fetch" error when the line "signifyClient.boot" got executed.

Reply:
another issue with just checking `signifyClient.boot` is that the boot might succeed but the connect fail

Reply:
> No, it could respond but with incorrect data. Failed to fetch means the port isn't open generally.

Oh, I got your idea now. But if it could response. How can we detect it as an error?

Reply:
I think this would still present itself as an error. When the Signify library receives a bad response from KERIA, it won't know how to continue so it should throw an error. So we just need to catch Signify throwing an error on those lines

Reply:
I think I will catch it if the error stack is related to `SignifyClient`. That's when the KERIA does not work properly.

Reply:
OK this works actually, thanks!